refactor(core): introduce tree-shakeable runtime error codes for `NgM…

SkyZeroZx opened the pull request:
…odule` handling and `ViewContainerRef` errors

Adds new tree-shakeable runtime error codes to improve error reporting for NgModule resolution issues (duplicate or missing IDs) and invalid ViewContainerRef operations involving destroyed views.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A

## What is the new behavior?

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

SkyZeroZx commented on packages/core/src/errors.ts:
```diff
@@ -135,6 +135,10 @@ export const enum RuntimeErrorCode {
   NO_COMPONENT_FACTORY_FOUND = 917,
   EXTERNAL_RESOURCE_LOADING_FAILED = 918,
   DEF_TYPE_UNDEFINED = -919,
+  NG_MODULE_ID_NOT_FOUND = 923,
```
Use code 923 since 922 will already be taken once PR #66015 is merged.

JeanMeche reviewed (approved):
reviewed-for: public-api

SkyZeroZx committed:
refactor(core): introduce tree-shakeable runtime error codes for `NgModule` handling and `ViewContainerRef` errors

Adds new tree-shakeable runtime error codes to improve error reporting for
NgModule resolution issues (duplicate or missing IDs) and invalid ViewContainerRef
operations involving destroyed views.

JeanMeche reviewed (approved):
I ran a TGP on this one, it's good to go.

kirjs commented:
This PR was merged into the repository. The changes were merged into the following branches:

- main: 1532be9d00fddc2bfe89cac2200e0d2a0a9fd4df
- 21.0.x: 8f26a52aa8f23ff335df004dfb091d59fdceacaf
